src/Sylius/Component/Core/Uploader/ImageUploader.php

```diff
@@ -71,6 +83,17 @@ public function remove(string $path): bool
         return false;
     }
 
+    private function getImagePath(ImageInterface $image): string
+    {
+        if (null !== $this->imagePathGenerator) {
+            return $this->imagePathGenerator->generate($image);
+        }
+
+        $hash = bin2hex(random_bytes(16));
+
+        return $this->expandPath($hash . '.' . $image->getFile()->guessExtension());
```

Comment:
The logic in these methods (`getImagePath` and `expandPath`) looks the same as in `UploadedImagePathGenerator`. This way we can create and assign `UploadedImagePathGenerator` instance in the constructor if no image path generator has been passed and get rid of this code in this class.